src/client/CourseRequestForm/index.js

```diff
@@ -73,48 +83,45 @@ const CourseRequestForm = ({ courses }) => {
         onChange={onChangeCallback}
       />
     ) : (
-      <Checkbox
-        label=""
-        size="small"
-        onChange={onChangeCallback}
-        defaultChecked
-      />
+      <Checkbox label="" size="small" onChange={onChangeCallback} />
     );
   const classTypeStyling = {
     ugrad: constants.ugradRow,
     grad: constants.gradRow,
     tut: constants.tutRow,
     none: constants.unselectedRow,
   };
-  const toggleToBeBuilt = (index) => {
-    coursesData[index].toBeBuilt = !coursesData[index].toBeBuilt;
-    setCoursesData(coursesData);
+  const toggleToBeBuilt = (courseCode) => {
+    const copy = new Set(tobeBuiltCheckedIds);
+    if (copy.has(courseCode)) {
+      copy.delete(courseCode);
+    } else {
+      copy.add(courseCode);
+    }
+
+    setTobeBuiltCheckedIds(copy);
   };
 
   const currentDate = new Date().toLocaleString();
 
   const courseListings = coursesData.map((course, index) =>
-    course.courseList.map((courseCode) => (
-      <Table.Row key={courseCode}>
-        <Table.Cell
-          theme={
-            classTypeStyling[
-              course.toBeBuilt === true ? course.classType : 'none'
-            ]
-          }
-        >
-          {index + 1}
-        </Table.Cell>
-        <Table.Cell
-          theme={
-            classTypeStyling[
-              course.toBeBuilt === true ? course.classType : 'none'
-            ]
-          }
-        >
-          {course.offeredTermCode}
-        </Table.Cell>
-        {/* <Table.Cell
+    course.courseList.map((courseCode) => {
+      const type = getCourseType(
+        course.courseCatalogNumber.substring(0, 4),
+        courseCode
+      );
+      return (
+        <Table.Row key={courseCode}>
+          <Table.Cell
+            theme={classTypeStyling[type]}
+            // Theme={classTypeStyling.getCourseType(course.courseCatalogNumber)}
+          >
+            {index + 1}
+          </Table.Cell>
+          <Table.Cell theme={classTypeStyling[type]}>
```

Comment:
@tatikamsetti Did setting theme at the Table.Row level not work?

Reply:
@rlorenzo Yeah it didn't somehow. I donno if it is being overridden somewhere! like... in the table header? I am not sure. I tried... but couldn't make it work.